contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -885,6 +917,243 @@ run_tsql_best_match_heuristics(int nargs, Oid *input_typeids, FuncCandidateList
 	return new_candidates;
 }
 
+/*
+ * getImmediateBaseTypeOfUDT()
+ * This funciton 
+ */
+static Oid getImmediateBaseTypeOfUDT(Oid typeid)
+{
+	Relation					relsetting;
+	HeapTuple					tuple;
+	ScanKeyData					scankey;
+	SysScanDesc					scan;
+	bool						isnull;
+	Datum						datum;
+	Datum                       tsql_typename;
+	LOCAL_FCINFO(fcinfo, 1);
+
+	/* if common_utility_plugin_ptr is not initialised */
+	if (common_utility_plugin_ptr == NULL)
+		ereport(ERROR,
+				(errcode(ERRCODE_INTERNAL_ERROR),
+					errmsg("Failed to find common utility plugin.")));
+
+	/* if tsql_typename is NULL it implies that inputTypId corresponds to UDT */
+	InitFunctionCallInfoData(*fcinfo, NULL, 0, InvalidOid, NULL, NULL);
+	fcinfo->args[0].value = ObjectIdGetDatum(typeid);
+	fcinfo->args[0].isnull = false;
+	tsql_typename = (*common_utility_plugin_ptr->translate_pg_type_to_tsql) (fcinfo);
+
+	/* if given type is not an UDT then return InvalidOid */
+	if (tsql_typename)
+		return InvalidOid;
+
+	/* Get immediate base type id of given type id */
+	relsetting = table_open(TypeRelationId, AccessShareLock);
+	ScanKeyInit(&scankey,
+				Anum_pg_type_oid,
+				BTEqualStrategyNumber, F_OIDEQ,
+				ObjectIdGetDatum(typeid));
+
+	scan = systable_beginscan(relsetting, TypeOidIndexId, true, NULL, 1, &scankey);
+	tuple = systable_getnext(scan);
+	if (HeapTupleIsValid(tuple))
+	{
+		datum = heap_getattr(tuple, Anum_pg_type_typbasetype, RelationGetDescr(relsetting), &isnull);
+		if (!isnull)
+		{
+			return DatumGetObjectId(datum);
+		}
+	}
+	systable_endscan(scan);
+	table_close(relsetting, AccessShareLock);
```

Comment:
Why all this? Can't we can just use `getBaseType` function?
IMO, we don't even need a separate function `getImmediateBaseTypeOfUDT` for this.

Reply:
Yeah, @sumitj824 is correct.